search-service/src/test/kotlin/com/egm/stellio/search/service/AttributeInstanceServiceTests.kt

```diff
@@ -332,6 +333,113 @@ class AttributeInstanceServiceTests : WithTimescaleContainer {
             .hasSize(3)
     }
 
+    @Test
+    fun `it should return min value of all instances for 30 day`() = runTest {
+        (1..9).forEachIndexed { index, _ ->
+            val attributeInstance =
+                gimmeAttributeInstance()
+                    .copy(
+                        measuredValue = index.toDouble(),
+                        time = ZonedDateTime.parse("2022-07-0${index + 1}T00:00:00Z")
+                    )
+            attributeInstanceService.create(attributeInstance)
+        }
+        val temporalQuery = TemporalQuery(
+            timerel = TemporalQuery.Timerel.AFTER,
+            timeAt = ZonedDateTime.parse("2022-07-03T00:00:00Z"),
+            timeBucket = "30 day",
+            aggregate = TemporalQuery.Aggregate.MIN
+        )
+
+        val enrichedEntity = attributeInstanceService.search(temporalQuery, temporalEntityAttribute, false)
+
+        assertThat(enrichedEntity)
+            .hasSize(1)
+            .singleElement()
```

Comment:
`.singleElement()` is a shorthand for `.hasSize(1).first()`, so you don't need the `hasSize(1)` above.